src/components/MapController/MapController.js

```diff
@@ -89,93 +157,220 @@ var MapController = React.createClass({
   },
 
   componentWillReceiveProps: function (nextProps) {
-    this.selectedDataset = nextProps.meta[0];
-
-    this.requestTimeMetadata(this.selectedDataset.unique_id).then(response => {
+    var defaultDataset = nextProps.meta[0];
+    var run = defaultDataset.ensemble_member;
+    var start_date = defaultDataset.start_date;
+    var end_date = defaultDataset.end_date;
+    
+    var times = {};
+    var timesPromises = [];
+    
+    //set colours. In order of preference:
+    //1. colours received by prop
+    //2. colours set by the user or this function previously and held in state
+    //3. defaults (scalar rainbow if a single dataset, 
+    //             scalar greyscale and contours rainbow for 2)
+    var sPalette, cPalette;
+    if(nextProps.scalarPalette) {
+      sPalette = nextProps.scalarPalette;
+      cPalette = nextProps.contourPalette;
+    }
+    else if(this.state.scalarPalette) {
+      sPalette = this.state.scalarPalette;
+      cPalette = this.state.contourPalette;
+    }
+    else {
+      sPalette = nextProps.comparandMeta ? 'seq-Greys' : 'x-Occam';
+      cPalette = nextProps.comparandMeta ? 'x-Occam': undefined;
+    }
```

Comment:
Hmm, a trifle tricky to understand. I'd go with a single if-else statement here instead of the two conditional assignments.